### PR TITLE
Add Doppler Database connection

### PR DESCRIPTION
**Pull request by nreal**
# Background
Send data to SAP with information getting from Doppler database
### Changes
- Add Dapper micro ORM

### Configuration
- Add connectionString as SercretKey

### Pending done
- Create SQL consult to get data
- Create an update SQL sentence to update with Doppler.Currency data
- Create service to send data of billing to SAP system

### CODE REVIEW

File: Doppler.Sap.Job/Database/DopplerRepository.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+using System.Threading.Tasks;
+using Dapper;
+
+namespace Doppler.Sap.Job.Service.Database
+{
+    public class DopplerRepository : IDopplerRepository
+    {
+        private readonly string _connectionString;
+
+        public DopplerRepository(string connectionString)
```
Comment (andresmoschini) [inline Doppler.Sap.Job/Database/DopplerRepository.cs:12]: How does DI container to know what string inject?

File: Doppler.Jobs.Server/Startup.cs [hunk 2]
```diff
@@ -74,6 +75,11 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddTransient<DopplerSapService>();
 
+            var connectionString = Configuration.GetConnectionString("DopplerDatabase");
+            services.AddSingleton(connectionString);
```
Comment (andresmoschini) [inline Doppler.Jobs.Server/Startup.cs:79]: What is the type of `connectionString`?

File: Doppler.Sap.Job/Settings/DopplerRepositorySettings.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Doppler.Sap.Job.Service.Settings
+{
+    public class DopplerRepositorySettings
+    {
+        public string ConnectionString { get; set; }
+    }
```
Comment (andresmoschini) [inline Doppler.Sap.Job/Settings/DopplerRepositorySettings.cs:6]: Is this class being used?

File: Doppler.Sap.Job/Database/IDopplerRepository.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Doppler.Sap.Job.Service.Database
+{
+    public interface IDopplerRepository
+    {
+        public Task<IEnumerable<object>> GetBillingClientInformation();
```
Comment (andresmoschini) [inline Doppler.Sap.Job/Database/IDopplerRepository.cs:9]: I would prefer a more descriptive type, even for initial steps.

File: Doppler.Jobs.Server/Startup.cs (hunk 2 above)
Comment (elsupergomez) [reply Doppler.Jobs.Server/Startup.cs:79]: Could be better use the already declared `DopplerRepositorySettings` class that hold the `ConnectionString` property. And use `IOption<T>` pattern could be improve this code.

Comment (nreal) [reply Doppler.Jobs.Server/Startup.cs]: > What is the type of `connectionString`?

I had an error between different approach, I fixed the problem with DopplerRepositorySettings class thanks @elsupergomez

File: Doppler.Sap.Job/Settings/DopplerRepositorySettings.cs (hunk 3 above)
Comment (nreal) [reply Doppler.Sap.Job/Settings/DopplerRepositorySettings.cs:6]: Now yes with this commit https://github.com/MakingSense/doppler-jobs/pull/6/commits/f5d1f949ed4d5c0dff5e14b70938148ea69b8c86

File: Doppler.Sap.Job/Database/DopplerRepository.cs (hunk 1 above)
Comment (nreal) [reply Doppler.Sap.Job/Database/DopplerRepository.cs]: I changed this code in this commit https://github.com/MakingSense/doppler-jobs/pull/6/commits/f5d1f949ed4d5c0dff5e14b70938148ea69b8c86

**Comment by fgchaio** [comment]
👍

File: Doppler.Sap.Job/Database/IDopplerRepository.cs (hunk 4 above)
Comment (nreal) [reply Doppler.Sap.Job/Database/IDopplerRepository.cs]: With @fgchaio we talk about it and I created a new `BillingCredit ` class in  this commit 
https://github.com/MakingSense/doppler-jobs/pull/6/commits/bfd4b80459426a2ca5d14619077e2202fb713732
Is possible that change this entity in the future

**Pull request merged**